packages/components/focus-scope/src/focus-scope.ts

```diff
@@ -0,0 +1,271 @@
+        }
+
+        // When the focused element gets removed from the DOM, browsers move focus
+        // back to the document.body. In this case, we move focus to the container
+        // to keep focus trapped correctly.
+        const handleMutations = (mutations: MutationRecord[]) => {
+          const focusedElement = document.activeElement as HTMLElement | null
+          if (focusedElement !== document.body)
+            return
+          for (const mutation of mutations) {
+            if (mutation.removedNodes.length > 0)
+              focus(container.value?.$el)
+          }
+        }
+
+        document.addEventListener('focusin', handleFocusIn)
+        document.addEventListener('focusout', handleFocusOut)
+        const mutationObserver: MutationObserver = new MutationObserver(handleMutations)
+        if (container.value)
+          mutationObserver.observe(container.value?.$el, { childList: true, subtree: true })
+        onInvalidate(() => {
+          document.removeEventListener('focusin', handleFocusIn)
+          document.removeEventListener('focusout', handleFocusOut)
+          mutationObserver.disconnect()
+        })
+      }
+    })
+
+    watchEffect((onInvalidate) => {
+      if (container.value) {
+        focusScopesStack.add(focusScope)
+        const previouslyFocusedElement = document.activeElement as HTMLElement | null
+        const hasFocusedCandidate = container.value?.$el.contains(previouslyFocusedElement)
+
+        if (!hasFocusedCandidate) {
+          const mountEvent = new CustomEvent(AUTOFOCUS_ON_MOUNT, EVENT_OPTIONS)
+          container.value?.$el.addEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus)
+          container.value?.$el.dispatchEvent(mountEvent)
+          if (!mountEvent.defaultPrevented) {
+            focusFirst(removeLinks(getTabbableCandidates(container.value?.$el)), { select: true })
+            if (document.activeElement === previouslyFocusedElement)
+              focus(container.value?.$el)
+          }
+        }
+        onInvalidate(() => {
+          container.value?.$el.removeEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus)
+
+          // We hit a react bug (fixed in v17) with focusing in unmount.
+          // We need to delay the focus a little to get around it for now.
+          // See: https://github.com/facebook/react/issues/17894
+          // setTimeout(() => {
+          const unmountEvent = new CustomEvent(AUTOFOCUS_ON_UNMOUNT, EVENT_OPTIONS)
+          container.value?.$el.addEventListener(AUTOFOCUS_ON_UNMOUNT, onUnmountAutoFocus)
+          container.value?.$el.dispatchEvent(unmountEvent)
+          if (!unmountEvent.defaultPrevented)
+            focus(previouslyFocusedElement ?? document.body, { select: true })
+          // we need to remove the listener after we `dispatchEvent`
+          container.value?.$el.removeEventListener(AUTOFOCUS_ON_UNMOUNT, onUnmountAutoFocus)
+
+          // focusScopesStack.remove(focusScope)
```

Comment:
This line is not working properly; it causes an infinite loop, and triggers the warning '[Vue warn]: Maximum recursive updates exceeded.'

As a consequence, unexpected behavior arises within the 'Multiple' and 'WithOptions' storybooks.

In the 'Multiple' scenario, when both traps are open, clicking on the outer input moves the focus event to it. Meanwhile, in the 'WithOptions' scenario, the 'age' field sometimes got focus, and other times it does not.

Could you please check it @productdevbook

Reply:
infitinite loop test 

This is the answer to the question here, I committed it.

```ts
  afterEach(() => wrapper.unmount());
```

age field focus bugs i see. I'm looking after

Reply:
No, that was in the vue playground. Because when I was running 'Muliable' story the browser crashed. It was not always crashing, there are steps if you follow the warning message will appear or the browser will crash.

- Open the Trap 1.
- Open the Trap 2.
- Close the Trap 1.
- Reopen the Trap 1.

Reply:
I don't see this problem, did you take the last commits and test them? me only problem `e 'age' field `

Reply:
Yeah, as I said the last commit fixed the first issue because you added the `await nextTick` to the second ```watchEffect```. The second issue that is related to the age field is not working and I still don't know why. 

I guess the watcher does not triggere when the emit update!

Reply:
```ts
@change="(event) => focusOnMount = (event.target as HTMLInputElement).checked ? ageFieldRef : true"
```

emit works, actually the problem is here, after the form v-if is removed, the element doesn't go and update it.

there is no problem with the core codes at the moment, the problem is on the demo side

Reply:
Do you mean that by removing the `v-if` the age field is getting focus?, becouse I tried before with `v-show` and it doesn't!

`v-if` is removing the the age field from the dom. So, if it works with you we need to add onMounted to the age `ref`.

Reply:
i am send new commit if 

```
  if (focusOnMount.value !== true) {
    event.preventDefault()
    if (ageFieldRef.value)
      ageFieldRef.value?.focus()
  }
  ```
  
change working but this is not similar to radix.

Reply:
I will check it when I reached home. I still need to move the exported type from the beginning of focus-scope file to the utilis file.

Reply:
https://vuejs.org/guide/essentials/reactivity-fundamentals.html#dom-update-timing

Reply:
if you think of something different you can post a new pr